JokeDAO/JokeDaoV2Dev
# feat: downvoting in UI and bytecode

# Description

- Add `downvote` option to contest creation form
- Add downvote button to proposal
-  Update cast vote modal UI to display "downvote" option (if the contest allows downvoting)
- Update ABI

## Type of change

- [x] New feature (breaking change)

# How Has This Been Tested?

Scenario 1: down vote enabled
1. Create a contest with "Downvoting" allowed :heavy_check_mark: 
2. Go to contest page, create  a  proposal and wait for votes to be open
3. You should see both upvote and downvote button appear
4. On "Up vote" click, the cast vote modal should open with the "up vote" option selected ; on "Down vote" click, the "down vote" option should be selected
5. Input X amount of vote
6. Click on "cast vote"
7. On transaction success, the number of votes available in your wallet should be updated and the number of votes on the proposal should be updated.

Scenario 1: down vote not enabled
1. Create a contest, don't allow downvotes
2. Go to contest page, create  a  proposal and wait for votes to be open
3. You should only see the upvote button appear
4. In the cast vote modal, no "upvote/downvote" options should appear



![Screenshot_20220811_145518](https://user-images.githubusercontent.com/15010369/184143742-759f0267-b426-40f2-94a9-2de77bbd4043.png)

![Screenshot_20220811_145125](https://user-images.githubusercontent.com/15010369/184143752-46f12d15-1cab-4821-9fe1-08ba305b646e.png)

![Screenshot_20220811_144952](https://user-images.githubusercontent.com/15010369/184143755-cde12f9d-6ed5-41e3-9e5e-
061c5e978cfb.png)

![Screenshot_20220811_144938](https://user-images.githubusercontent.com/15010369/184143758-c4b4a503-ec58-49d6-89ee-b5ded1e8b362.png)

![Screenshot_20220811_144920](https://user-images.githubusercontent.com/15010369/184143762-0a916328-f392-4010-8d18-fd331c754a46.png)

![Screenshot_20220811_144900](https://user-images.githubusercontent.com/15010369/184143763-df61f65a-499d-4edb-9ee0-caf13c277704.png)

![Screenshot_20220811_144837](https://user-images.githubusercontent.com/15010369/184143765-df6f6e1d-2606-4d85-bd70-43a7ba68e9c1.png)
